src/components/Arpscan/Arpscan.tsx

```diff
@@ -0,0 +1,208 @@
+     * @param {string} data - The data received from the child process.
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data);
+    }, []);
+
+    /**
+     * handleProcessTermination: Callback to handle the termination of the child process.
+     * Once the process termination is handled, it clears the process PID reference and
+     * deactivates the loading overlay.
+     * @param {object} param - An object containing information about the process termination.
+     * @param {number} param.code - The exit code of the terminated process.
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            // If the process was successful, display a success message.
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+
+                // If the process was terminated manually, display a termination message.
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+
+                // If the process was terminated with an error, display the exit and signal codes.
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference. There is no longer a valid process running.
+            setPid("");
+
+            // Cancel the loading overlay. The process has completed.
+            setLoading(false);
+
+            // Now that loading has completed, allow the user to save the output to a file.
+            setAllowSave(true);
+            setHasSaved(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    /**
+     * handleSaveComplete: Recognizes that the output file has been saved.
+     * Passes the saved status back to SaveOutputToTextFile_v2.
+     */
+    const handleSaveComplete = () => {
+        setHasSaved(true);
+        setAllowSave(false);
+    };
+
+    /**
+     * onSubmit: Asynchronous handler for the form submission event.
+     * It sets up and triggers the ArpScan tool with the given parameters.
+     * Once the command is executed, the results or errors are displayed in the output.
+     * @param {FormValuesType} values - The form values, containing the network interface.
+     */
+    const onSubmit = async (values: FormValuesType) => {
+        // Set the loading state to true to indicate that the process is starting.
+        setLoading(true);
+
+        // Disable saving the output to a file while the process is running.
+        setAllowSave(false);
+
+        // Construct the arguments for the ArpScan command.
+        const args = ["-I", values.interface];
```

Comment:
Are you sure this is the right options? When I try in the terminal I get the following:

![image](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/assets/35479648/b843f0b3-f512-43bc-8fe8-e76c2f13f8a4)

- [ ] Check to make sure the right options are being used.

Reply:
We just updated the code.